src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +17,34 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(movie);
+            transaction.commit();
+        } catch (IllegalArgumentException e) {
+            throw new DataProcessingException("The instance of Movie is not an entity.", e);
+        } catch (IllegalStateException | PersistenceException e) {
```

Comment:
Consider using more specific exceptions for better clarity. `IllegalArgumentException` might not be the best choice for handling non-entity instances. Additionally, `IllegalStateException` is not typically associated with persistence operations, so it might be unnecessary here.